docs/sources/flow/reference/components/loki.process.md

```diff
@@ -0,0 +1,191 @@
+to match the type of stage to configure. Multiple processing stages must be
+defined in different blocks.
+
+The following arguments are supported:
+
+Name                  | Type          | Description                               | Default        | Required
+--------------------- | --------------| ----------------------------------------- | -------------- | --------
+`labels`              | `map(string)` | Configures a `labels` processing stage.   | `{}`           | no
+
+
+The `labels` argument configures a [Labels processing stage][] that can read data
+from the extracted values map and set new labels on incoming log entries.
+
+[Labels processing stage]: #labels-stage
+
+
+### json block
+
+The `json` inner block configures a [JSON Processing stage][] that parses incoming
+log lines or previously extracted values as JSON and uses
+[JMESPath expressions](https://jmespath.org/tutorial.html) to extract new
+values from them.
+
+Name             | Type          | Description | Default | Required
+---------------- | ------------- | ----------- | ------- | --------
+`expressions`    | `map(string)` | Key-value pairs of JMESPath expressions. | | yes
+`source`         | `string`      | Source of the data to parse as JSON. | `""` | no
+`drop_malformed` | `bool`        | Drop lines whose input cannot be parsed as valid JSON.| `false` | no
+
+[JSON Processing stage]: #json-stage
+
+
+
+## Exported fields
+
+The following fields are exported and can be referenced by other components:
+
+Name | Type | Description
+---- | ---- | -----------
+`receiver` | `LogsReceiver` | A value that other components can use to send log entries to.
+
+## Component health
+
+`loki.process` is only reported as unhealthy if given an invalid
+configuration.
+
+## Debug information
+
+`loki.process` does not expose any component-specific debug
+information.
+
+## Debug metrics
+* `loki_process_dropped_lines_total` (counter): Number of lines dropped as part of a processing stage.
+
+## Example
+
+This example creates a `loki.process` component that extracts the `environment`
+value from a JSON log line and sets it as a label.
+```river
+loki.process "local" {
+  forward_to = [loki.write.onprem.receiver]
+
+  stage {
+    json {
+      expressions = { "extracted_env" = "environment" }
+    }
+  }
+
+  stage {
+    labels = { "env" = "extracted_env" }
+  }
+}
+```
+
+## JSON stage
+When configuring a JSON stage, the `source` field defines the source of data to
+parse as JSON. By default, this is the log line itself, but it can also be a
+previously extracted value.
+
+The `expressions` field is the set of key-value pairs of MESPath expressions to
+run. The map key defines the name with which the data is extracted as, while
+the map value is the expression used to populate the value.
+
+<!-- TODO(@tpaschalis) Should we move this example here to a different tutorial?
+I don't feel the fields are enough for someone to understand what the processing
+stage is doing -->
```

Comment:
Yeah, I think examples should be more copy-and-paste focused rather than explaining details. I think explanations belong either closer to where the thing is originally defined (i.e., in the "json block" header) or in a tutorial where we can dive deeper.

Reply:
Okay, I'll move them back to each subblock definition, hope it doesn't make them too verbose.